cocotb/share/makefiles/Makefile.inc

```diff
@@ -57,9 +57,18 @@ else ifneq (, $(findstring MSYS, $(OS)))
 endif
 export OS
 
+# Detects if Python is running in a virtual environment
+# https://docs.python.org/3/library/venv.html
+VENV=$(shell $(shell cocotb-config --python-bin) -c 'import sys; print(sys.prefix != sys.base_prefix)')
+
 # this ensures we use the same python as the one cocotb was installed into
-# realpath to convert windows paths to unix paths, like cygpath -u
-PYTHON_BIN ?= $(realpath $(shell cocotb-config --python-bin))
+ifeq ($(VENV),True)
+    # In a virtual environment, the Python binary may be a symlink, so it should not use realpath
+    PYTHON_BIN ?= $(shell cocotb-config --python-bin)
+else
+    # realpath to convert windows paths to unix paths, like cygpath -u
+    PYTHON_BIN ?= $(realpath $(shell cocotb-config --python-bin))
+endif
```

Comment:
```suggestion
PYTHON_BIN ?= $(realpath -s $(shell cocotb-config --python-bin))
```

Reply:
This makes `PYTHON_BIN` point *outside* of the virtual environment, and breaks how virtual environments work. Are you trying to avoid problems on Windows? If so, what about making this conditional on Windows?

Reply:
I think what you have works fine. No need for an OS conditional if we can help it.

Reply:
Is not that `-s` for `realpath` should do?
```
       -s, --strip, --no-symlinks
              don't expand symlinks
```

Reply:
I missed the `-s` in your first comment, but I tried it and it still expands symlinks, at least on macOS. Perhaps because the macOS version of the `realpath` command doesn't have a `--strip` option?

There is `$(abspath ...)`, but that may be [GNU Make specific](https://www.gnu.org/software/make/manual/html_node/File-Name-Functions.html), and not portable?

What problem is this trying to solve? Using whatever `python3` is in the `$PATH` without mucking with it seems like the most portable solution. Yet it still allows users to customize the binary by setting their `$PATH` correctly, similarly to how scripts start with `#!/usr/bin/env`.

Reply:
No issues just adding `-s` seems nicer to me. 

>macOS version of the realpath command doesn't have a --strip option?

There is no end to this madness.